Modules/threadmodule.c

```diff
@@ -904,6 +904,10 @@ initthread(void)
 {
     PyObject *m, *d;
 
+    if (PyErr_WarnPy3k_WithFix("In 3.x, the thread module is removed", 
```

Comment:
Is the `threading` module compatible with `thread`? In other words, if I e.g. turn `import thread.X` into `import thread.X` will that work? I wonder if we need more fine-grained warnings?

Reply:
Yes, good one, that is possible, we can warn both ways. For module, and the method where the warning is more fine grained e.g "thread.get_indent was removed in 3.x: use threading._ge_indent instead"

Also note even if you use the threading module, some methods were renamed to have an underscore e.g `thread.get_indent` is `threading._get_indent` there is no way to write code that works on both 2.x and 3.x for like 3 methods

Reply:
Is the `threading` module in Python 2 compatible with Python 3? I'm wondering if we have to do some warnings in pygrate2 and some others in pygrate3?

Reply:
Not completely, some methods are named with an underscore in 2.x and yet in 3.x they dont have the underscore.

Reply:
3 methods are completely incompatible

Reply:
This then sounds like something where we can easily provide good quality warnings?

Reply:
Yes, do we need to forward port too?  its as simple as aliasing the incompatible methods

Reply:
IMHO, if the user can change things in pygrate2, we shouldn't warn in pygrate3. But if they can't change in pygrate2, we should consider aliases & warnings in pygrate3.

Reply:
OK SGTM

Reply:
Should we do that in this PR? [I tend to think so, but I don't have a firm opinion.]

Reply:
Onmy radar this morning, I want to handle as part of this PR